internal/remotewrite/proxy.go

```diff
@@ -57,41 +59,72 @@ func remoteWrite(write *url.URL, endpoints []Endpoint, logger log.Logger) http.H
 		}
 
 		rlogger := log.With(logger, "request", middleware.GetReqID(r.Context()))
-		for _, endpoint := range endpoints {
+		for i, endpoint := range endpoints {
 			var client *http.Client
 			var err error
 			if endpoint.ClientConfig == nil {
 				client = &http.Client{}
 			} else {
-				client, err = promconfig.NewClientFromConfig(*endpoint.ClientConfig, endpoint.Name, true)
+				client, err = promconfig.NewClientFromConfig(*endpoint.ClientConfig, endpoint.Name,
+					promconfig.WithDialContextFunc((&net.Dialer{
+						Timeout:   30 * time.Second,
+						KeepAlive: 30 * time.Second,
+					}).DialContext))
 				if err != nil {
-					level.Error(rlogger).Log("failed to create a new HTTP client", "err", err)
+					//level.Error(rlogger).Log("msg", "failed to create a new HTTP client", "err", err)
+					LogChannels[i] <- logMessage{
+						messageKey: "failed to create a new HTTP client",
+						keyvals: []interface{}{
+							"msg", "failed to create a new HTTP client", "err", err,
+						}}
 				}
 			}
 
 			req, err := http.NewRequest(http.MethodPost, endpoint.URL, bytes.NewReader(body))
 			req.Header = r.Header
 			if err != nil {
-				level.Error(rlogger).Log("msg", "Failed to create the forward request", "err", err, "url", endpoint.URL)
+				//level.Error(rlogger).Log("msg", "Failed to create the forward request", "err", err, "url", endpoint.URL)
+				LogChannels[i] <- logMessage{
+					messageKey: "Failed to create the forward request",
```

Comment:
Suggest to use lower case for all messages to comply with the golang standard.

Reply:
done